Index: evolution-python/src/ebook.defs
===================================================================
--- evolution-python.orig/src/ebook.defs	2007-07-20 15:10:48.000000000 -0300
+++ evolution-python/src/ebook.defs	2007-07-20 15:13:53.000000000 -0300
@@ -167,6 +167,15 @@
   )
 )
 
+(define-method commit_contact
+  (of-object "EBook")
+  (c-name "e_book_commit_contact")
+  (return-type "gboolean")
+  (parameters
+    '("EContact*" "contact")
+  )
+)
+
 
 ;;------------------------------------------------------------------------------
 ;; evo-environment
Index: evolution-python/src/ebook.override
===================================================================
--- evolution-python.orig/src/ebook.override	2007-07-20 15:10:57.000000000 -0300
+++ evolution-python/src/ebook.override	2007-07-20 15:21:37.000000000 -0300
@@ -169,4 +169,36 @@
 
     return PyLong_FromLong((long) seconds);
 }
+%%
+override e_book_commit_contact kwargs
+static PyObject *
+_wrap_e_book_commit_contact(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] = {"contact", NULL};
+    PyGObject *pycontact;
+    PyObject *pyret;
+    EBook *book;
+    EContact *contact;
+    gboolean ret;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!:EBook.commit_contact",
+                                     kwlist, &PyEContact_Type,
+                                     (PyObject **)&pycontact)){
+        return NULL;
+    }
+
+    contact = (EContact *)pycontact->obj;
+    book = (EBook *)self->obj;
+
+    ret = e_book_commit_contact(book, contact, NULL);
+
+    if (ret) {
+        pyret = Py_True;
+    } else {
+        pyret = Py_False;
+    }
+
+    Py_XINCREF(pyret);
+    return pyret;
+}
 
