Index: evolution-python-0.0.1/src/Makefile.am
===================================================================
--- evolution-python-0.0.1.orig/src/Makefile.am	2007-06-20 16:11:19.000000000 -0300
+++ evolution-python-0.0.1/src/Makefile.am	2007-06-20 16:11:50.000000000 -0300
@@ -19,6 +19,9 @@
 
 nodist_evolution_la_SOURCES = evolution.c
 
+defsdir = $(pkgdatadir)/defs
+defs_DATA = evolution.defs
+
 CLEANFILES = evolution.c
 EXTRA_DIST = \
 	evolution.defs \
