Index: evolution-python/src/ebook.defs
===================================================================
--- evolution-python.orig/src/ebook.defs	2007-07-23 15:08:15.000000000 -0300
+++ evolution-python/src/ebook.defs	2007-07-23 15:10:03.000000000 -0300
@@ -23,6 +23,13 @@
   (gtype-id "E_TYPE_BOOK")
 )
 
+(define-object EBookView
+  (in-module "EBook")
+  (parent "GObject")
+  (c-name "EBookView")
+  (gtype-id "E_TYPE_BOOK_VIEW")
+)
+
 ;;------------------------------------------------------------------------------
 ;; EVContact
 ;;------------------------------------------------------------------------------
@@ -322,6 +329,27 @@
 )
 
 ;;------------------------------------------------------------------------------
+;; EBookView
+;;------------------------------------------------------------------------------
+
+(define-function e_book_view_get_type
+  (c-name "e_book_view_get_type")
+  (return-type "GType")
+)
+
+(define-method start
+  (of-object "EBookView")
+  (c-name "e_book_view_start")
+  (return-type "none")
+)
+
+(define-method stop
+  (of-object "EBookView")
+  (c-name "e_book_view_stop")
+  (return-type "none")
+)
+
+;;------------------------------------------------------------------------------
 ;; evo-environment
 ;;------------------------------------------------------------------------------
 (define-function list_addressbooks
