Index: evolution-python/src/ebook.defs
===================================================================
--- evolution-python.orig/src/ebook.defs	2007-07-23 16:52:45.000000000 -0300
+++ evolution-python/src/ebook.defs	2007-07-23 16:52:46.000000000 -0300
@@ -315,6 +315,20 @@
   )
 )
 
+; New get_book_view
+(define-method get_book_view
+  (of-object "EBook")
+  (c-name "e_book_get_book_view")
+  (return-type "gboolean")
+  (parameters
+    '("EBookQuery*" "query")
+    '("GList*" "requested_fields")
+    '("int" "max_results")
+    '("EBookView**" "book_view")
+    '("GError**" "error")
+  )
+)
+
 ;; EBookQuery enumerations
 (define-enum EBookQueryTest
   (in-module "evolution")
@@ -349,6 +363,51 @@
   (return-type "none")
 )
 
+(define-method set_contacts_added_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_set_contacts_added_cb")
+  (return-type "int")
+)
+
+(define-method remove_contacts_added_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_remove_contacts_added_cb")
+  (parameters
+    '("int", "handler_id")
+  )
+  (return-type "none")
+)
+
+(define-method set_contacts_changed_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_set_contacts_changed_cb")
+  (return-type "int")
+)
+
+(define-method remove_contacts_changed_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_remove_contacts_changed_cb")
+  (parameters
+    '("int", "handler_id")
+  )
+  (return-type "none")
+)
+
+(define-method set_contacts_removed_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_set_contacts_removed_cb")
+  (return-type "int")
+)
+
+(define-method remove_contacts_removed_cb
+  (of-object "EBookView")
+  (c-name "e_book_view_remove_contacts_removed_cb")
+  (parameters
+    '("int", "handler_id")
+  )
+  (return-type "none")
+)
+
 ;;------------------------------------------------------------------------------
 ;; evo-environment
 ;;------------------------------------------------------------------------------
Index: evolution-python/src/ebook.override
===================================================================
--- evolution-python.orig/src/ebook.override	2007-07-23 16:52:45.000000000 -0300
+++ evolution-python/src/ebook.override	2007-07-23 18:08:35.000000000 -0300
@@ -425,6 +425,99 @@
 	return pyebook_query_new(e_book_query_copy(self->query));
 }
 
+/*** Types, helper functions and callbacks for EBookView ***/
+
+/* Struct used to pack data passed to EBookView signals */
+struct bookview_cb_t{
+    PyObject *(*converter)(gpointer);
+    PyObject *callback;
+    PyObject *extra_data;
+};
+
+static PyObject *
+glist_to_pylist(GList *list, PyObject *(*converter)(gpointer))
+{
+    PyObject *pylist;
+    GList *node;
+    int size;
+    int i;
+
+    size = g_list_length(list);
+    pylist = PyList_New(size);
+    node = list;
+
+    for (i=0; i < size; i++){
+        PyObject *pyobj;
+
+        pyobj = converter(node->data);
+        PyList_SetItem(pylist, i, pyobj);
+
+        node = node->next;
+    }
+
+    return pylist;
+}
+
+PyObject *econtact_to_pycontact(EContact *contact)
+{
+    return pygobject_new(G_OBJECT(contact));
+}
+
+/* wrapper for callbacks that receive a GList as parameter */
+static void
+contacts_list_cb(EBookView *ebookview, GList *contacts, gpointer user_data)
+{
+    PyObject *list;
+    PyObject *pybookview;
+    PyObject *extra_args;
+    PyObject *callback;
+    PyObject *arguments;
+    PyObject *result;
+    struct bookview_cb_t *data;
+    int i;
+    int extra_size;
+
+    pybookview = pygobject_new(ebookview);
+    if (pybookview == NULL){
+    }
+
+    data = (struct bookview_cb_t*)user_data;
+
+    callback = data->callback;
+    extra_args = data->extra_data;
+
+    list = glist_to_pylist(contacts, data->converter);
+
+    if (extra_args != NULL){
+        extra_size = PyTuple_Size(extra_args);
+    } else {
+        extra_size = 0;
+    }
+
+    arguments = PyTuple_New(extra_size+2);
+
+    Py_INCREF(pybookview);
+    PyTuple_SetItem(arguments, 0, pybookview);
+    Py_INCREF(list);
+    PyTuple_SetItem(arguments, 1, list);
+
+    for (i = 2; i < extra_size+2; i++) {
+        PyObject *obj;
+
+        obj = PyTuple_GetItem(extra_args, i-2);
+        Py_INCREF(obj);
+        PyTuple_SetItem(arguments, i, obj);
+    }
+
+    result = PyObject_CallObject(callback, arguments);
+
+    Py_XDECREF(result);
+    Py_XDECREF(arguments);
+    Py_XDECREF(pybookview);
+    Py_XDECREF(list); /* Will decref its childs too? */
+
+    return;
+}
 
 %%
 init
@@ -590,4 +683,239 @@
     Py_XINCREF(pyret);
     return pyret;
 }
+%%
+override e_book_get_book_view kwargs
+static PyObject *
+_wrap_e_book_get_book_view(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] = {"query", "fields", "max", NULL};
+    int results = 0;
+    PyObject *pyfields = NULL;
+    GList *fields = NULL;
+    PyObject *query = NULL;
+    PyObject *ret = NULL;
+    int cret;
+    EBookView *view = NULL;
+    EBookQuery *equery = NULL;
+    GError *err = NULL;
+
+    if(!PyArg_ParseTupleAndKeywords(args, kwargs, "O!|O!i:EBook.get_book_view",
+                                    kwlist, &PyEBookQuery_Type, &query, &PyList_Type, &pyfields,
+                                    &results))
+        return NULL;
+
+    if (pyfields != NULL) {
+        /* Translating to glist */
+        int len;
+        int i;
+
+        len = PyList_Size(pyfields);
+
+        for(i = 0; i < len; i++){
+            PyObject *item;
+            EContactField field;
+            char *name;
+            int size;
+
+            item = PyList_GetItem(pyfields, i);
+
+            pyg_enum_get_value(E_TYPE_CONTACT_FIELD, item, (gint *)&field);
+            name = e_contact_field_name(field);
+            fields = g_list_append(fields, name);
+        }
+    }
+
+    equery = ((PyEBookQuery *)query)->query;
+    cret = e_book_get_book_view(E_BOOK(self->obj), equery, fields, results, &view, &err);
+
+    /* Create the bookview object */
+    ret = pygobject_new(G_OBJECT(view));
+
+    /* e_book_query_unref(equery); The book view gets the ownership of the ref */
+    if (fields != NULL)
+        g_list_free(fields);
+
+    Py_XINCREF(ret);
+    return ret;
+}
+%%
+override e_book_view_set_contacts_added_cb kwargs
+static PyObject *
+_wrap_e_book_view_set_contacts_added_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *callback;
+    PyObject *extra_args;
+    struct bookview_cb_t *data;
+    int len;
+    guint result;
+
+    len = PyTuple_Size(args);
+    if (len < 1){
+        PyErr_SetString(PyExc_TypeError,
+                        "EBookView.set_contacts_added_cb requires at least one argument");
+        return NULL;
+    }
+
+    data = g_new(struct bookview_cb_t, 1);
+
+    callback = PyTuple_GetItem(args, 0);
+
+    if(!PyCallable_Check(callback)){
+        PyErr_SetString(PyExc_TypeError, "callback must be callable");
+        return NULL;
+    }
+
+    Py_XINCREF(callback);
+    data->callback = callback;
+
+    if (len > 1){
+        extra_args = PySequence_GetSlice(args, 1, len);
+    } else {
+        extra_args = NULL;
+    }
+
+    data->extra_data = extra_args;
+    data->converter = econtact_to_pycontact;
+
+    result = g_signal_connect(self->obj, "contacts-added", contacts_list_cb, data);
+
+    return PyInt_FromLong(result);
+}
+%%
+override e_book_view_remove_contacts_added_cb kwargs
+static PyObject *
+_wrap_e_book_view_remove_contacts_added_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] ={"handler_id", NULL};
+    int handler;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "i", kwlist, &handler)) {
+        return NULL;
+    }
+
+    g_signal_handler_disconnect(self->obj, handler);
+
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+%%
+override e_book_view_set_contacts_changed_cb kwargs
+static PyObject *
+_wrap_e_book_view_set_contacts_changed_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *callback;
+    PyObject *extra_args;
+    struct bookview_cb_t *data;
+    int len;
+    guint result;
+
+    len = PyTuple_Size(args);
+    if (len < 1){
+        PyErr_SetString(PyExc_TypeError,
+                        "EBookView.set_contacts_added_cb requires at least one argument");
+        return NULL;
+    }
+
+    data = g_new(struct bookview_cb_t, 1);
+
+    callback = PyTuple_GetItem(args, 0);
+
+    if(!PyCallable_Check(callback)){
+        PyErr_SetString(PyExc_TypeError, "callback must be callable");
+        return NULL;
+    }
+
+    Py_XINCREF(callback);
+    data->callback = callback;
+
+    if (len > 1){
+        extra_args = PySequence_GetSlice(args, 1, len);
+    } else {
+        extra_args = NULL;
+    }
+
+    data->extra_data = extra_args;
+    data->converter = econtact_to_pycontact;
+
+    result = g_signal_connect(self->obj, "contacts-changed", contacts_list_cb, data);
+
+    return PyInt_FromLong(result);
+}
+%%
+override e_book_view_remove_contacts_changed_cb kwargs
+static PyObject *
+_wrap_e_book_view_remove_contacts_changed_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] ={"handler_id", NULL};
+    int handler;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "i", kwlist, &handler)) {
+        return NULL;
+    }
+
+    g_signal_handler_disconnect(self->obj, handler);
+
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+%%
+override e_book_view_set_contacts_removed_cb kwargs
+static PyObject *
+_wrap_e_book_view_set_contacts_removed_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *callback;
+    PyObject *extra_args;
+    struct bookview_cb_t *data;
+    int len;
+    guint result;
+
+    len = PyTuple_Size(args);
+    if (len < 1){
+        PyErr_SetString(PyExc_TypeError,
+                        "EBookView.set_contacts_removed_cb requires at least one argument");
+        return NULL;
+    }
+
+    data = g_new(struct bookview_cb_t, 1);
+
+    callback = PyTuple_GetItem(args, 0);
+
+    if(!PyCallable_Check(callback)){
+        PyErr_SetString(PyExc_TypeError, "callback must be callable");
+        return NULL;
+    }
+
+    Py_XINCREF(callback);
+    data->callback = callback;
+
+    if (len > 1){
+        extra_args = PySequence_GetSlice(args, 1, len);
+    } else {
+        extra_args = NULL;
+    }
+
+    data->extra_data = extra_args;
+    data->converter = PyString_FromString;
+
+    result = g_signal_connect(self->obj, "contacts-removed", contacts_list_cb, data);
+
+    return PyInt_FromLong(result);
+}
+%%
+override e_book_view_remove_contacts_removed_cb kwargs
+static PyObject *
+_wrap_e_book_view_remove_contacts_removed_cb(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] ={"handler_id", NULL};
+    int handler;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "i", kwlist, &handler)) {
+        return NULL;
+    }
+
+    g_signal_handler_disconnect(self->obj, handler);
+
+    Py_INCREF(Py_None);
+    return Py_None;
+}
 
Index: evolution-python/test/bookview.py
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ evolution-python/test/bookview.py	2007-07-23 18:03:54.000000000 -0300
@@ -0,0 +1,215 @@
+#!/usr/bin/python2.5
+
+import time
+import unittest
+
+import evolution.ebook as e
+import gobject
+
+class TestBookView(unittest.TestCase):
+
+    def setUp(self):
+        self.loop = gobject.MainLoop()
+        self.context = self.loop.get_context()
+        self.book = e.open_addressbook("default")
+        self.contact_id = None # Id of created contacts for cleanup
+        self.contacts = None # Contact list from callbacks
+
+    def testCommit(self):
+        c1 = e.EContact()
+        c1.props.full_name = "Original Name"
+        id1 = self.book.add_contact(c1)
+
+        c2 = self.book.get_contact(id1)
+        c2.props.full_name = "Modified Name"
+        self.book.commit_contact(c2)
+
+        c3 = self.book.get_contact(id1)
+        self.assertEqual(c2.props.full_name, c3.props.full_name)
+
+        self.contact_id = id1
+
+
+    def testContactsAddedCb_NoArgs(self):
+        query = e.EBookQuery.field_test(e.CONTACT_FULL_NAME,
+                                        e.BOOK_QUERY_CONTAINS,
+                                        "Dummy")
+        view = self.book.get_book_view(query)
+
+        h1 = view.set_contacts_added_cb(self.added_cb)
+        view.start()
+
+        c1 = e.EContact()
+        c1.props.full_name = "Dummy Contact"
+
+        self.book.add_contact(c1) # Generates the signal
+        self.contact_id = c1.props.id
+        self.refresh_gui()
+
+        if (self.contacts):
+            self.assertEqual(self.contacts[0].props.full_name,
+                             c1.props.full_name)
+            self.assertEqual(self.contacts[0].props.id,
+                             c1.props.id)
+        else:
+            raise TypeError("Contacts not updated")
+
+        view.remove_contacts_added_cb(h1)
+
+    def testContactsAddedCb_Args(self):
+        extra_data = "Extra Data"
+        query = e.EBookQuery.field_test(e.CONTACT_FULL_NAME,
+                                        e.BOOK_QUERY_CONTAINS,
+                                        "Dummy")
+
+        view = self.book.get_book_view(query)
+
+        h1 = view.set_contacts_added_cb(self.added_cb, extra_data)
+        view.start()
+
+        c1 = e.EContact()
+        c1.props.full_name = "Dummy Contact"
+
+        self.book.add_contact(c1)
+        self.contact_id = c1.props.id
+        self.refresh_gui()
+
+        if (self.contacts):
+            self.assertEqual(self.contacts[0].props.full_name,
+                             c1.props.full_name)
+            self.assertEqual(extra_data, self.extra_data)
+        else:
+            raise TypeError("Contacts not updated")
+
+        view.remove_contacts_added_cb(h1)
+
+    def testContactsChangedCb_NoArgs(self):
+
+        new_name = "Dummy Modified"
+
+        query = e.EBookQuery.field_test(e.CONTACT_FULL_NAME,
+                                        e.BOOK_QUERY_CONTAINS,
+                                        "Dummy")
+        view = self.book.get_book_view(query)
+
+        c1 = e.EContact()
+        c1.props.full_name = "Dummy Contact"
+
+        id1 = self.book.add_contact(c1)
+        self.contact_id = id1
+
+        h1 = view.set_contacts_changed_cb(self.added_cb)
+        view.start()
+
+        c2 = self.book.get_contact(id1)
+        c2.props.full_name = new_name
+        self.book.commit_contact(c2) # Generates the signal
+
+        self.refresh_gui()
+
+        if (self.contacts):
+            self.assertEqual(self.contacts[0].props.full_name,
+                             new_name)
+            self.assertEqual(self.contacts[0].props.id,
+                             c2.props.id)
+        else:
+            raise TypeError("Contacts not updated")
+
+        view.remove_contacts_changed_cb(h1)
+
+    def testContactsChangedCb_Args(self):
+        extra_data = "Extra Data"
+        new_name = "Dummy Modified"
+        query = e.EBookQuery.field_test(e.CONTACT_FULL_NAME,
+                                        e.BOOK_QUERY_CONTAINS,
+                                        "Dummy")
+
+        view = self.book.get_book_view(query)
+
+        c1 = e.EContact()
+        c1.props.full_name = "Dummy Contact"
+        h1 = view.set_contacts_changed_cb(self.added_cb, extra_data)
+        view.start()
+
+        id1 = self.book.add_contact(c1)
+        self.contact_id = c1.props.id
+
+        c2 = self.book.get_contact(id1)
+        c2.props.full_name = new_name
+
+        self.book.commit_contact(c2) # Generates the signal
+        self.refresh_gui()
+
+        if (self.contacts):
+            self.assertEqual(self.contacts[0].props.full_name,
+                             new_name)
+            self.assertEqual(extra_data, self.extra_data)
+        else:
+            raise TypeError("Contacts not updated")
+
+        view.remove_contacts_changed_cb(h1)
+
+    def testGetBookViewWithFields(self):
+        query = e.EBookQuery.field_test(e.CONTACT_FULL_NAME,
+                                        e.BOOK_QUERY_CONTAINS,
+                                        "Dummy")
+
+        print query
+
+        view = self.book.get_book_view(query, [e.CONTACT_NICKNAME])
+        view.start()
+
+        print view
+        self.refresh_gui()
+
+    def added_cb(self, view, contacts, extra_data=None):
+        self.contacts = contacts
+        self.extra_data = extra_data
+
+    def tearDown(self):
+        if self.contact_id:
+            self.book.remove_contact_by_id(self.contact_id)
+
+        self.loop.quit()
+        del(self.loop)
+        del(self.book)
+
+    def refresh_gui(self, delay=0):
+        while self.context.pending():
+            self.context.iteration()
+        time.sleep(delay)
+
+    def build_view(self, name, query):
+        c1 = e.EContact()
+        c1.props.full_name = name
+        view = self.book.get_book_view(query)
+
+        return (view, c1)
+
+
+def added_cb(view, contacts, *user_data):
+    print view
+    print contacts
+    print user_data
+
+def main():
+    book = e.open_addressbook("default")
+
+    query = e.EBookQuery.field_exists(e.CONTACT_FULL_NAME)
+
+    view = book.get_book_view(query)
+    handler = view.set_contacts_added_cb(added_cb)
+    h2 = view.set_contacts_added_cb(added_cb, "one")
+    h3 = view.set_contacts_removed_cb(added_cb, "removed", "extra_args")
+    h4 = view.set_contacts_changed_cb(added_cb, "changed")
+    #print view.set_contacts_added_cb(added_cb, None)
+
+    view.remove_contacts_added_cb(handler)
+
+    view.start()
+
+    gtk.main()
+
+
+if __name__ == "__main__":
+    unittest.main()
