Index: evolution-python-0.0.1/src/evolution.defs
===================================================================
--- evolution-python-0.0.1.orig/src/evolution.defs	2007-07-10 15:56:07.000000000 -0300
+++ evolution-python-0.0.1/src/evolution.defs	2007-07-10 15:56:16.000000000 -0300
@@ -312,6 +312,15 @@
   )
 )
 
+(define-method commit_contact
+  (of-object "EBook")
+  (c-name "e_book_commit_contact")
+  (return-type "gboolean")
+  (parameters
+    '("EContact*" "contact")
+  )
+)
+
 ;; EBookQuery enumerations
 (define-enum EBookQueryTest
   (in-module "evolution")
Index: evolution-python-0.0.1/src/evolution.override
===================================================================
--- evolution-python-0.0.1.orig/src/evolution.override	2007-07-10 15:56:07.000000000 -0300
+++ evolution-python-0.0.1/src/evolution.override	2007-07-10 15:56:16.000000000 -0300
@@ -744,9 +744,6 @@
     data->extra_data = extra_args;
     data->converter = econtact_to_pycontact;
 
-    g_debug("structure passed: %p{%p,%p,%p}", data, data->converter,
-                                              data->callback, data->extra_data);
-
     /*Py_XINCREF(args);*/
     result = g_signal_connect(self->obj, "contacts-added", contacts_list_cb, data);
 
@@ -808,11 +805,6 @@
     data->extra_data = extra_args;
     data->converter = econtact_to_pycontact;
 
-    /*
-    g_debug("structure passed: %p{%p,%p,%p}", data, data->converter,
-                                              data->callback, data->extra_data);*/
-
-    /*Py_XINCREF(args);*/
     result = g_signal_connect(self->obj, "contacts-changed", contacts_list_cb, data);
 
     return PyInt_FromLong(result);
@@ -894,4 +886,31 @@
     Py_INCREF(Py_None);
     return Py_None;
 }
+%%
+override e_book_commit_contact kwargs
+static PyObject *
+_wrap_e_book_commit_contact(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] = {"contact", NULL};
+    PyGObject *pycontact;
+    PyObject *pyret;
+    EContact *contact;
+    gboolean ret;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!:EBook.commit_contact",
+                                     kwlist, &PyEContact_Type,
+                                     (PyObject **)&pycontact)){
+        return NULL;
+    }
 
+    ret = e_book_commit_contact(self->obj, pycontact->obj, NULL);
+
+    if (ret) {
+        pyret = Py_True;
+    } else {
+        pyret = Py_False;
+    }
+
+    Py_XINCREF(pyret);
+    return pyret;
+}
