Index: galago-python-0.5.0/debian/changelog
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ galago-python-0.5.0/debian/changelog	2007-08-21 09:59:25.000000000 -0300
@@ -0,0 +1,6 @@
+galago-python (0.5.0-1osso1) unstable; urgency=low
+
+  * First python-galago release.
+
+ -- Lauro Moura Maranhao Neto <lauro.neto@indt.org.br>  Thu, 09 Aug 2007 17:05:00 -0300
+
Index: galago-python-0.5.0/debian/compat
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ galago-python-0.5.0/debian/compat	2007-08-21 09:59:25.000000000 -0300
@@ -0,0 +1 @@
+4
Index: galago-python-0.5.0/debian/control
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ galago-python-0.5.0/debian/control	2007-08-21 09:59:25.000000000 -0300
@@ -0,0 +1,17 @@
+Source: galago-python
+Priority: optional
+Section: python
+Maintainer: Lauro Moura Mara
+Build-Depends: debhelper (>= 4.0.0), python2.5-gtk2-dev (>= 2.10.3-1osso4), python2.5-dev (>= 2.5.1-1osso2), libgalago-dev (>= 0.5.0)
+Standards-Version: 3.6.1
+
+Package: python2.5-galago
+Architecture: any
+Depends: python2.5-gtk2 (>= 2.10.3-1osso4), python2.5 (>= 2.5.1-1osso2), libgalago1 (>= 0.5.0) ${shlibs:Depends}
+Description: Python bindings for Galago library
+
+Package: python2.5-galago-dev
+Architecture: all
+Depends: python2.5-galago (= ${Source-Version})
+Description: Examples and header files for Python bindings for Galago library
+
Index: galago-python-0.5.0/debian/copyright
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ galago-python-0.5.0/debian/copyright	2007-08-21 09:59:25.000000000 -0300
@@ -0,0 +1,12 @@
+This package was debianized by Lauro Moura Maranhao Neto <lauro.neto@indt.org.br> on
+Thu, 09 Aug 2007 16:45:17 -0300.
+
+It was downloaded from http://pymaemo.garage.maemo.org
+
+Copyright:
+
+Upstream Author(s): Christian Hammond <chipx86@chipx86.com>
+
+License:
+
+See COPYING file.
Index: galago-python-0.5.0/debian/rules
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ galago-python-0.5.0/debian/rules	2007-08-21 09:59:25.000000000 -0300
@@ -0,0 +1,83 @@
+#!/usr/bin/make -f
+
+PYVER=2.5
+PYTHON=python$(PYVER)
+
+p_base=$(PYTHON)-galago
+p_dev=$(PYTHON)-galago-dev
+d_base=$(CURDIR)/debian/$(p_base)
+d_dev=$(CURDIR)/debian/$(p_dev)
+
+ifeq ($(DEB_BUILD_ARCH),armel)
+    CFLAGS=-Os -mthumb -mfloat-abi=softfp -Wall -Wstrict-prototypes
+else
+    CFLAGS=-Os -Wall -Wstrict-prototypes
+endif
+
+
+config.status: configure
+	dh_testdir
+	CFLAGS="$(CFLAGS)" PYTHON="$(PYTHON)" ./autogen.sh \
+		   --prefix=/usr \
+		   --mandir=\$${prefix}/share/man \
+		   --infodir=\$${prefix}/share/info
+
+build: build-stamp
+build-stamp:  config.status
+	dh_testdir
+	$(MAKE)
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+	-$(MAKE) distclean
+	rm -rf $(d_base) $(d_dev)
+	find . -name "*.pyo" | xargs rm -f
+	find . -name "*.pyc" | xargs rm -f
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+	$(MAKE) install DESTDIR=$(d_base)
+	$(PYTHON) -m compileall -f $(d_base)/usr/lib/$(PYTHON)
+	$(PYTHON) -OO -m compileall -f $(d_base)/usr/lib/$(PYTHON)
+
+binary-indep: build install
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installdocs -p$(p_dev)
+	dh_installchangelogs -p$(p_dev) ChangeLog
+
+	mkdir -p $(d_dev)
+	cp -a $(d_base)/usr $(d_dev)
+
+	# base
+	find $(d_base) -name "*.py" | xargs rm -f
+	find $(d_base) -name "*.pyc" | xargs rm -f
+	find $(d_base) -name "*.la" | xargs rm -f
+	rm -rf $(d_base)/usr/share $(d_base)/usr/lib/pkgconfig
+
+	# dev
+	find $(d_dev) -name "*.pyo" -or -name "*.so" | xargs rm -f
+	#rm $(d_dev)/usr/lib/$(PYTHON)/site-packages/pygst.pth
+	#mv $(d_dev)/usr/share/gst-python/0.10/examples $(d_dev)/usr/share/doc/$(p_dev)
+
+	dh_link
+	dh_strip
+	dh_compress -Xexample
+	dh_fixperms
+	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
