Index: trunk/debian/changelog
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/debian/changelog	2007-05-20 00:18:02.000000000 +0300
@@ -0,0 +1,346 @@
+ipython (0.8.1-1.maemo2) unstable; urgency=low
+
+  * bump revision
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Wed, 31 Oct 2007 11:02:52 -0300
+
+ipython (0.8.1-1.maemo1) unstable; urgency=low
+
+  * upgrade to v0.8.1
+
+ -- Ed Bartosh <bartosh@gmail.com>  Sun, 20 May 2007 00:17:22 +0300
+
+ipython (0.7.3-2.maemo3) unstable; urgency=low
+
+  * join pymaemo: use quilt instead of dpatch
+
+ -- Ed Bartosh <bartosh@gmail.com>  Sun,  1 Apr 2007 17:01:02 +0300
+
+ipython (0.7.3-2.maemo2) unstable; urgency=low
+
+  * libreadline4 added to dependency list 
+
+ -- Ed Bartosh <bartosh@gmail.com>  Sun, 18 Mar 2007 12:46:37 +0200
+
+ipython (0.7.3-2.maemo1) unstable; urgency=low
+
+  * Maemo build
+  * ipmaker.py: Add _Helper class (not present in site.py from python2.5) 
+
+ -- Ed Bartosh <bartosh@gmail.com>  Sun,  4 Mar 2007 00:39:39 +0200
+
+ipython (0.7.3-2) unstable; urgency=low
+
+  * Added a new patch from svn to fix jobctrl to work properly on posix.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Thu, 21 Dec 2006 20:13:57 +0100
+
+ipython (0.7.3-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Mon, 18 Dec 2006 22:23:55 +0100
+
+ipython (0.7.3~rc2-1) experimental; urgency=low
+
+  * New upstream release candidate.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sat, 16 Dec 2006 02:20:13 +0100
+
+ipython (0.7.3~beta2-1) experimental; urgency=low
+
+  * New upstream beta release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Fri,  8 Dec 2006 08:02:16 +0100
+
+ipython (0.7.3~beta1-1) experimental; urgency=low
+
+  * New upstream beta release.
+  * Removed backported patch added in 0.7.2-5 to workaround bugs in python
+    2.3's inspect module.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Wed, 29 Nov 2006 12:35:22 +0100
+
+ipython (0.7.2-6) UNRELEASED; urgency=low
+
+  * Added XS-Vcs-Svn field.
+
+ -- Piotr Ozarowski <ozarow@gmail.com>  Thu, 23 Nov 2006 14:44:43 +0100
+
+ipython (0.7.2-5) unstable; urgency=low
+
+  * Added a new patch from svn to workaround bugs in python 2.3's inspect
+    module. (closes: #374625)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Thu, 10 Aug 2006 18:36:12 +0200
+
+ipython (0.7.2-4) unstable; urgency=low
+
+  * Fixed spelling error in description. (closes: #363976)
+  * Ack NMU 0.7.2-3.1, thanks Matthias. (closes: #377787)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Tue,  1 Aug 2006 22:45:11 +0200
+
+ipython (0.7.2-3.1) unstable; urgency=medium
+
+  * NMU.
+  * Convert to updated Python policy. (closes: #377787)
+
+ -- Matthias Klose <doko@debian.org>  Thu, 13 Jul 2006 17:42:06 +0000
+
+ipython (0.7.2-3ubuntu1) edgy; urgency=low
+
+  * Synchronize with Debian unstable.
+  * Convert to updated Python policy. Collapse all packages into one
+    ipython package, don't handle ipython using alternatives.
+
+ -- Matthias Klose <doko@ubuntu.com>  Tue, 11 Jul 2006 09:47:37 +0000
+
+ipython (0.7.2-3) unstable; urgency=low
+
+  * Removed alternative for irunner manpage.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sat, 17 Jun 2006 09:49:10 +0200
+
+ipython (0.7.2-2) unstable; urgency=medium
+
+  * Fixed conflict in irunner. (closes: #373874)
+  * Added recommendation for python-pexpect. (closes: #373794)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Fri, 16 Jun 2006 10:43:45 +0200
+
+ipython (0.7.2-1) unstable; urgency=low
+
+  [ Piotr Ozarowski ]
+  * Added watch file.
+
+  [ Norbert Tretkowski ]
+  * New upstream release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Thu,  8 Jun 2006 23:36:03 +0200
+
+ipython (0.7.1.fix1+0.7.2.rc1-1) experimental; urgency=low
+
+  * New upstream release candidate.
+  * Updated Standards-Version to 3.7.2.0, no changes required.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sat, 27 May 2006 14:49:24 +0200
+
+ipython (0.7.1.fix1-2) unstable; urgency=low
+
+  * Set maintainer to Debian Python modules team and added myself to
+    uploaders.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sun, 16 Apr 2006 15:53:43 +0200
+
+ipython (0.7.1.fix1-1) unstable; urgency=low
+
+  * New upstream bugfix release.
+  * Removed backported patch which was added in 0.7.1-3 to catch
+    KeyboardInterrupt exceptions properly, it's part of this release.
+  * Fixed names of pdfs in doc-base file to shut up linda.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Tue, 14 Feb 2006 23:51:17 +0100
+
+ipython (0.7.1-3) unstable; urgency=low
+
+  * Added a new patch from upstream to catch KeyboardInterrupt exceptions
+    properly.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Mon, 30 Jan 2006 19:42:31 +0100
+
+ipython (0.7.1-2) unstable; urgency=low
+
+  * Really remove alternatives on purge, thanks Lars Wirzenius for finding
+    the problem. (closes: #317269)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sun, 29 Jan 2006 23:11:28 +0100
+
+ipython (0.7.1-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Tue, 24 Jan 2006 21:42:33 +0100
+
+ipython (0.7.0-2) unstable; urgency=low
+
+  * Fixed circular dependencies (closes: #341980)
+  * Added version to dependency on ipython dummy package. (closes: #320235)
+  * Removed python2.2 package, ipython now depends on python >= 2.3.
+  * Bumped up standards-version, no changes needed.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sat, 21 Jan 2006 23:27:53 +0100
+
+ipython (0.7.0-1) unstable; urgency=low
+
+  * New upstream release.
+  * Updated 01_docdir-base.dpatch and 02_profiler-message.dpatch.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sat, 21 Jan 2006 20:08:23 +0100
+
+ipython (0.6.15-2) unstable; urgency=low
+
+  * New maintainer, thanks Jack for your work.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sun, 28 Aug 2005 19:57:09 +0200
+
+ipython (0.6.15-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Thu,  2 Jun 2005 23:51:45 +0200
+
+ipython (0.6.14-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Tue, 31 May 2005 22:53:25 +0200
+
+ipython (0.6.13-1) unstable; urgency=low
+
+  * New upstream release.
+  * Removed backported patch which was added in 0.6.12-3 to fix misleading
+    prompt, it's part of this release.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Fri, 15 Apr 2005 09:42:35 +0200
+
+ipython (0.6.12-4) unstable; urgency=medium
+
+  * Re-added python build-dependency, it got lost in 0.6.12-2.
+    (closes: #301636)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sun, 27 Mar 2005 14:28:26 +0200
+
+ipython (0.6.12-3) unstable; urgency=low
+
+  * Added a new patch from cvs to fix misleading prompt2. (closes: #300847)
+
+ -- Norbert Tretkowski <nobse@debian.org>  Sun, 27 Mar 2005 00:05:26 +0100
+
+ipython (0.6.12-2) unstable; urgency=low
+
+  * Added packages for python2.2 and python2.4, ipython package is now a dummy
+    package depending on ipython built for Debians default python.
+    (closes: #292537)
+  * Split out generic files into separate ipython-common package.
+  * Enhanced package descriptions.
+  * Removed CFLAGS settings from debian/rules, not required.
+  * Tweaked message displayed when profiler support is missing.
+  * Suggest the python-profiler package.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Fri, 25 Mar 2005 20:24:36 +0100
+
+ipython (0.6.12-1) unstable; urgency=low
+
+  * New upstream release.
+  * Removed patch which was added in 0.6.5-1.1 to make profiling support
+    optional, it was merged upstream.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Wed,  2 Mar 2005 12:15:09 +0100
+
+ipython (0.6.11-1) unstable; urgency=low
+
+  * New upstream release.
+    + Fixed broken profiling support unless -D is specified. (closes: #295779)
+  * Acknowledged NMUs. (closes: #206653, #294500, #294861, #280505)
+  * New co-maintainer, added myself to uploaders.
+
+ -- Norbert Tretkowski <nobse@debian.org>  Tue,  1 Mar 2005 12:40:33 +0100
+
+ipython (0.6.5-1.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Rebuild with a python version that is actually in Debian.
+
+ -- Wichert Akkerman <wichert@wiggy.net>  Thu, 17 Feb 2005 23:08:52 +0100
+
+ipython (0.6.5-1.1) unstable; urgency=low
+
+  * NMU to apply patch making profiling support optional (provided by
+    Torsten Marek). (closes: #294500)
+
+ -- Steven R. Baker <srbaker@debian.org>  Thu, 17 Feb 2005 05:02:55 -0400
+
+ipython (0.6.5-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Jack Moffitt <jack@xiph.org>  Thu,  2 Dec 2004 15:49:27 -0700
+
+ipython (0.6.4-1.1) unstable; urgency=low
+
+  * NMU from BSP Frankfurt:
+    - Added Build-Depends on dpatch (Closes: #280505)
+
+ -- Joerg Jaspert <joerg@debian.org>  Sat, 27 Nov 2004 18:28:17 +0100
+
+ipython (0.6.4-1) unstable; urgency=low
+
+  * New upstream release
+  * Updated debian/rules to use dpatch and added debian/patches/*
+
+ -- Jack Moffitt <jack@xiph.org>  Tue,  9 Nov 2004 10:38:51 -0700
+
+ipython (0.6.3-1) unstable; urgency=low
+
+  * New upstream release (Closes: #262244, #252525)
+
+ -- Jack Moffitt <jack@xiph.org>  Wed, 22 Sep 2004 21:39:38 -0600
+
+ipython (0.6.0-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Jack Moffitt <jack@xiph.org>  Sun,  9 May 2004 17:38:00 +0200
+
+ipython (0.4.0-1.1) unstable; urgency=low
+
+  * Non maintainer upload with maintainer consensus.
+  * Updated buil-dep to depend on python-2.3 (Closes: #206653)
+  * Included encoding comment string in modules under IPython to 
+    avoid python2.3 Warnigs (upstream noticed).
+
+ -- Marco Presi (Zufus) <zufus@debian.org>  Mon, 25 Aug 2003 19:02:20 +0200
+
+ipython (0.4.0-1) unstable; urgency=low
+
+  * New upstream release (Closes #195215)
+  * Updated Build-Depends (Closes #200021)
+
+ -- Jack Moffitt <jack@xiph.org>  Fri, 25 Jul 2003 10:16:12 -0600
+
+ipython (0.2.15pre3-4) unstable; urgency=low
+
+  * Add python-dev Build-Depends (Closes: #189901)
+
+ -- Jack Moffitt <jack@babyjesus.cantcode.com>  Mon, 12 May 2003 23:33:43 -0600
+
+ipython (0.2.15pre3-3) unstable; urgency=low
+
+  * Fix for accidently converting home directory files' line ends
+    instead of just the appropriate ipython configuration files
+    (Closes: #189042)
+
+ -- Jack Moffitt <jack@xiph.org>  Mon, 14 Apr 2003 21:18:18 -0600
+
+ipython (0.2.15pre3-2) unstable; urgency=low
+
+  * Added text of the PSF license to debian/copyright.
+
+ -- Jack Moffitt <jack@xiph.org>  Sat,  5 Apr 2003 11:41:31 -0700
+
+ipython (0.2.15pre3-1) unstable; urgency=low
+
+  * Updated documentation to point to the new homepage -
+    http://ipython.scipy.org
+  * Removed doc/COPYING and doc/GNU-LGPL to fix lintian warnings
+  * Public release ready (Closes: #185789)
+
+ -- Jack Moffitt <jack@xiph.org>  Tue,  1 Apr 2003 20:53:31 -0700
+
+ipython (0.2.15pre3-0a) unstable; urgency=low
+
+  * Initial Release.
+  * This is my first Debian package
+
+ -- Jack Moffitt <jack@xiph.org>  Wed, 12 Mar 2003 21:04:22 -0700
+
Index: trunk/debian/compat
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/debian/compat	2007-05-20 00:15:35.000000000 +0300
@@ -0,0 +1 @@
+4
Index: ipython-0.8.1/debian/control
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/debian/control	2007-05-20 00:15:35.000000000 +0300
@@ -0,0 +1,27 @@
+Source: ipython
+Section: User/python
+Priority: optional
+Maintainer: Ed Bartosh <bartosh@gmail.com>
+Section: user/programming
+Build-Depends-Indep: debhelper (>= 4), quilt (>= 0.45), python2.5-dev (>= 2.5.0-1osso10)
+Standards-Version: 3.7.2.1
+
+Package: ipython
+Architecture: all
+Depends: python2.5 (>= 2.5.0-1osso10), libreadline4
+Recommends: python-numeric, python-numeric-ext, python-pexpect
+Conflicts: python2.3-ipython, python2.4-ipython, ipython-common
+Replaces: python2.3-ipython, python2.4-ipython, ipython-common
+Suggests: python-profiler
+Description: enhanced interactive Python shell
+ IPython is an enhanced interactive Python shell. It can be used as a
+ replacement for the standard Python shell, or it can be used as a
+ complete working environment for scientific computing (like Matlab or
+ Mathematica) when paired with the standard Python scientific and
+ numerical tools. It supports dynamic object introspections, numbered
+ input/output prompts, a macro system, session logging, session
+ restoring, complete system shell access, verbose and colored traceback
+ reports, auto-parentheses, auto-quoting, and is embeddable in other
+ Python programs.
+ .
+ Homepage: http://ipython.scipy.org/ 
Index: trunk/debian/copyright
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/debian/copyright	2007-05-20 00:15:35.000000000 +0300
@@ -0,0 +1,168 @@
+This package was debianized by Jack Moffitt <jack@xiph.org> on
+Wed, 12 Mar 2003 20:38:14 -0700.
+
+It was downloaded from http://ipython.scipy.org/
+
+Upstream Author: Fernando Perez <fperez@colorado.edu>,
+		 Janko Hauser <jhauser@ifm.uni-kiel.de>,
+		 Nathaniel Gray <n8gray@caltech.edu>
+
+Copyright:
+
+Most IPython code is copyright (C) 2001 by Fernando Perez, Janko
+Hauser, and Nathaniel Gray.  All code is licensed under the GNU Lesser
+General Public License (LGPL) except as explicitly mentioned below.
+Its full text is included in the file /usr/share/common-licenses/LGPL.
+
+DPyGetOpt.py is copyright (C) 2001 by Bill Bumgarner <bbum@friday.com>
+and is licensed under the MIT license, reproduced below:
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to
+deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+sell copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+IN THE SOFTWARE.
+
+FlexCompleter.py is copyright (C) 2001 by the Python Software
+Foundation and licensed under the terms of the Python Software
+Foundation License, reproduced below:
+
+A. HISTORY OF THE SOFTWARE
+==========================
+
+Python was created in the early 1990s by Guido van Rossum at Stichting
+Mathematisch Centrum (CWI) in the Netherlands as a successor of a
+language called ABC.  Guido is Python's principal author, although it
+includes many contributions from others.  The last version released
+from CWI was Python 1.2.  In 1995, Guido continued his work on Python
+at the Corporation for National Research Initiatives (CNRI) in Reston,
+Virginia where he released several versions of the software.  Python
+1.6 was the last of the versions released by CNRI.  In 2000, Guido and
+the Python core development team moved to BeOpen.com to form the
+BeOpen PythonLabs team.  Python 2.0 was the first and only release
+from BeOpen.com.
+
+Following the release of Python 1.6, and after Guido van Rossum left
+CNRI to work with commercial software developers, it became clear that
+the ability to use Python with software available under the GNU Public
+License (GPL) was very desirable.  CNRI and the Free Software
+Foundation (FSF) interacted to develop enabling wording changes to the
+Python license.  Python 1.6.1 is essentially the same as Python 1.6,
+with a few minor bug fixes, and with a different license that enables
+later versions to be GPL-compatible.  Python 2.1 is a derivative work
+of Python 1.6.1, as well as of Python 2.0.
+
+After Python 2.0 was released by BeOpen.com, Guido van Rossum and the
+other PythonLabs developers joined Digital Creations.  All
+intellectual property added from this point on, starting with Python
+2.1 and its alpha and beta releases, is owned by the Python Software
+Foundation (PSF), a non-profit modeled after the Apache Software
+Foundation.  See http://www.python.org/psf/ for more information about
+the PSF.
+
+Thanks to the many outside volunteers who have worked under Guido's
+direction to make these releases possible.
+
+
+B. TERMS AND CONDITIONS FOR ACCESSING OR OTHERWISE USING PYTHON
+===============================================================
+
+PSF LICENSE AGREEMENT
+---------------------
+
+1. This LICENSE AGREEMENT is between the Python Software Foundation
+("PSF"), and the Individual or Organization ("Licensee") accessing and
+otherwise using Python 2.1 software in source or binary form and its
+associated documentation.
+
+2. Subject to the terms and conditions of this License Agreement, PSF
+hereby grants Licensee a nonexclusive, royalty-free, world-wide
+license to reproduce, analyze, test, perform and/or display publicly,
+prepare derivative works, distribute, and otherwise use Python 2.1
+alone or in any derivative version, provided, however, that PSF's
+License Agreement and PSF's notice of copyright, i.e., "Copyright (c)
+2001 Python Software Foundation; All Rights Reserved" are retained in
+Python 2.1 alone or in any derivative version prepared by Licensee.
+
+3. In the event Licensee prepares a derivative work that is based on
+or incorporates Python 2.1 or any part thereof, and wants to make
+the derivative work available to others as provided herein, then
+Licensee hereby agrees to include in any such work a brief summary of
+the changes made to Python 2.1.
+
+4. PSF is making Python 2.1 available to Licensee on an "AS IS"
+basis.  PSF MAKES NO REPRESENTATIONS OR WARRANTIES, EXPRESS OR
+IMPLIED.  BY WAY OF EXAMPLE, BUT NOT LIMITATION, PSF MAKES NO AND
+DISCLAIMS ANY REPRESENTATION OR WARRANTY OF MERCHANTABILITY OR FITNESS
+FOR ANY PARTICULAR PURPOSE OR THAT THE USE OF PYTHON 2.1 WILL NOT
+INFRINGE ANY THIRD PARTY RIGHTS.
+
+5. PSF SHALL NOT BE LIABLE TO LICENSEE OR ANY OTHER USERS OF PYTHON
+2.1 FOR ANY INCIDENTAL, SPECIAL, OR CONSEQUENTIAL DAMAGES OR LOSS AS A
+RESULT OF MODIFYING, DISTRIBUTING, OR OTHERWISE USING PYTHON 2.1, OR
+ANY DERIVATIVE THEREOF, EVEN IF ADVISED OF THE POSSIBILITY THEREOF.
+
+6. This License Agreement will automatically terminate upon a material
+breach of its terms and conditions.
+
+7. This License Agreement shall be governed by the federal
+intellectual property law of the United States, including without
+limitation the federal copyright law, and, to the extent such
+U.S. federal law does not apply, by the law of the Commonwealth of
+Virginia, excluding Virginia's conflict of law provisions.
+Notwithstanding the foregoing, with regard to derivative works based
+on Python 1.6.1 that incorporate non-separable material that was
+previously distributed under the GNU General Public License (GPL), the
+law of the Commonwealth of Virginia shall govern this License
+Agreement only as to issues arising under or with respect to
+Paragraphs 4, 5, and 7 of this License Agreement.  Nothing in this
+License Agreement shall be deemed to create any relationship of
+agency, partnership, or joint venture between CNRI and Licensee.  This
+License Agreement does not grant permission to use CNRI trademarks or
+trade name in a trademark sense to endorse or promote products or
+services of Licensee, or any third party.
+
+8. By clicking on the "ACCEPT" button where indicated, or by copying,
+installing or otherwise using Python 1.6.1, Licensee agrees to be
+bound by the terms and conditions of this License Agreement.
+
+        ACCEPT
+
+
+CWI PERMISSIONS STATEMENT AND DISCLAIMER
+----------------------------------------
+
+Copyright (c) 1991 - 1995, Stichting Mathematisch Centrum Amsterdam,
+The Netherlands.  All rights reserved.
+
+Permission to use, copy, modify, and distribute this software and its
+documentation for any purpose and without fee is hereby granted,
+provided that the above copyright notice appear in all copies and that
+both that copyright notice and this permission notice appear in
+supporting documentation, and that the name of Stichting Mathematisch
+Centrum or CWI not be used in advertising or publicity pertaining to
+distribution of the software without specific, written prior
+permission.
+
+STICHTING MATHEMATISCH CENTRUM DISCLAIMS ALL WARRANTIES WITH REGARD TO
+THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
+FITNESS, IN NO EVENT SHALL STICHTING MATHEMATISCH CENTRUM BE LIABLE
+FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT
+OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+Itpl.py is copyright (C) 2001 Ka-Ping Yee <ping@lfw.org> and licensed
+under the MIT license reproduced above.
Index: trunk/debian/rules
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/debian/rules	2007-05-20 00:15:35.000000000 +0300
@@ -0,0 +1,116 @@
+#!/usr/bin/make -f
+# ipython debian/rules file
+#
+# Based on sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+PYDEF := 2.3
+PYVERS := 2.5
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+
+	set -e; \
+	for v in $(PYVERS); do \
+	  python$$(echo $$v | sed "s/$(PYDEF)//") setup.py config; \
+	done
+
+	touch configure-stamp
+
+
+build: configure build
+
+build-stamp:
+	dh_testdir
+
+	set -e; \
+	for v in $(PYVERS); do \
+	  python$$(echo $$v | sed "s/$(PYDEF)//") setup.py build; \
+	done
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	set -e; \
+	for v in $(PYVERS); do \
+	  python$$v setup.py clean --all; \
+	done
+	rm -rf build
+	rm -f setupext/*.py?
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	set -e; \
+	for v in $(PYVERS); do \
+	  python$$(echo $$v | sed "s/$(PYDEF)//") setup.py install --prefix $(CURDIR)/debian/ipython/usr; \
+	done
+
+	# remove documentation
+	rm $(CURDIR)/debian/ipython/usr/share/doc/ipython/COPYING
+	rm $(CURDIR)/debian/ipython/usr/share/doc/ipython/ChangeLog
+	rm $(CURDIR)/debian/ipython/usr/share/doc/ipython/ipython.el
+	rm $(CURDIR)/debian/ipython/usr/share/doc/ipython/README_Windows.txt
+	rm $(CURDIR)/debian/ipython/usr/share/doc/ipython/pycon.ico
+
+	# change permission on scripts
+	chmod a-x $(CURDIR)/debian/ipython/usr/share/doc/ipython/examples/*
+
+ifeq (0,1)
+	# rename binaries
+	mv $(CURDIR)/debian/python2.5-ipython/usr/bin/ipython $(CURDIR)/debian/python2.5-ipython/usr/bin/python2.5-ipython
+	mv $(CURDIR)/debian/python2.5-ipython/usr/bin/pycolor $(CURDIR)/debian/python2.5-ipython/usr/bin/python2.5-pycolor
+	mv $(CURDIR)/debian/python2.5-ipython/usr/bin/irunner $(CURDIR)/debian/python2.5-ipython/usr/bin/python2.5-irunner
+
+	# rename manpages
+	mv $(CURDIR)/debian/python2.5-ipython/usr/share/man/man1/ipython.1.gz $(CURDIR)/debian/python2.5-ipython/usr/share/man/man1/python2.5-ipython.1.gz
+	mv $(CURDIR)/debian/python2.5-ipython/usr/share/man/man1/pycolor.1.gz $(CURDIR)/debian/python2.5-ipython/usr/share/man/man1/python2.5-pycolor.1.gz
+endif
+
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_link
+	dh_installchangelogs doc/ChangeLog
+	dh_installdocs
+#	dh_installexamples
+	dh_install
+#	dh_installmenu
+#	dh_installdebconf	
+#	dh_installlogrotate
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installmime
+#	dh_installinit
+#	dh_installcron
+#	dh_installinfo
+#	dh_installman doc/ipython.1.gz doc/pycolor.1.gz
+	dh_compress -Xexamples
+	dh_fixperms
+	dh_python
+#	dh_makeshlibs
+	dh_installdeb
+#	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+# Build architecture-dependent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
