Index: trunk/IPython/ipmaker.py
===================================================================
--- trunk.orig/IPython/ipmaker.py	2007-04-01 16:24:07.000000000 +0300
+++ trunk/IPython/ipmaker.py	2007-04-01 16:24:31.000000000 +0300
@@ -58,6 +58,18 @@
 from IPython.usage import cmd_line_usage,interactive_usage
 from IPython.genutils import *
 
+
+class _Helper:
+    def __repr__(self):
+        return "Type help() for interactive help, " \
+               "or help(object) for help about object."
+    def __call__(self, *args, **kwds):
+        import pydoc
+        return pydoc.help(*args, **kwds)
+ 
+__builtin__.help = _Helper()
+
+
 #-----------------------------------------------------------------------------
 def make_IPython(argv=None,user_ns=None,user_global_ns=None,debug=1,
                  rc_override=None,shell_class=InteractiveShell,
@@ -95,7 +107,6 @@
                      embedded=embedded,**kw)
 
     # Put 'help' in the user namespace
-    from site import _Helper
     IP.user_ns['help'] = _Helper()
 
 
