Index: pygame/examples/aliens.py
===================================================================
--- pygame.orig/examples/aliens.py	2007-03-06 16:34:35.000000000 -0300
+++ pygame/examples/aliens.py	2007-03-06 16:34:55.000000000 -0300
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.5
 
 import random, os.path
 
Index: pygame/examples/arraydemo.py
===================================================================
--- pygame.orig/examples/arraydemo.py	2007-03-06 16:34:35.000000000 -0300
+++ pygame/examples/arraydemo.py	2007-03-06 16:34:55.000000000 -0300
@@ -18,7 +18,7 @@
 
 def surfdemo_show(array_img, name):
     "displays a surface, waits for user to continue"
-    screen = pygame.display.set_mode(array_img.shape[:2], 0, 32)
+    screen = pygame.display.set_mode(array_img.shape[:2], 0, 16)
     surfarray.blit_array(screen, array_img)
     pygame.display.flip()
     pygame.display.set_caption(name)
Index: pygame/examples/fonty.py
===================================================================
--- pygame.orig/examples/fonty.py	2007-03-06 16:34:35.000000000 -0300
+++ pygame/examples/fonty.py	2007-03-06 16:34:55.000000000 -0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.5
 
 """Here we load a .TTF font file, and display it in
 a basic pygame window. It demonstrates several of the
Index: pygame/examples/stars.py
===================================================================
--- pygame.orig/examples/stars.py	2007-03-06 16:34:35.000000000 -0300
+++ pygame/examples/stars.py	2007-03-06 16:34:55.000000000 -0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.5
 
 """A simple starfield example. Note you can move the 'center' of
 the starfield by leftclicking in the window. This example show
