Index: python-setuptools-0.6c6/setuptools/command/sdist.py
===================================================================
--- python-setuptools-0.6c6.orig/setuptools/command/sdist.py	2006-10-31 14:20:42.000000000 -0300
+++ python-setuptools-0.6c6/setuptools/command/sdist.py	2008-09-11 16:28:32.000000000 -0300
@@ -1,5 +1,6 @@
 from distutils.command.sdist import sdist as _sdist
 from distutils.util import convert_path
+from distutils import log
 import os, re, sys, pkg_resources
 
 entities = [
