Index: gnome-python-2.18.0/gnome/__init__.py
===================================================================
--- gnome-python-2.18.0.orig/gnome/__init__.py	2007-01-08 09:07:04.000000000 -0300
+++ gnome-python-2.18.0/gnome/__init__.py	2008-03-31 20:25:00.000000000 -0300
@@ -1,15 +1,4 @@
 # -*- Mode: Python; py-indent-offset: 4 -*-
 
 # this can go when things are a little further along
-try:
-    import ltihooks
-    del ltihooks
-except ImportError:
-    pass
-
-import gobject
-del gobject
-
-from _gnome import *
-init = program_init
-
+pass
