Index: gnome-python-2.18/examples/vfs/async-xfer.py
===================================================================
--- gnome-python-2.18.orig/examples/vfs/async-xfer.py	2007-04-04 17:27:07.000000000 -0300
+++ gnome-python-2.18/examples/vfs/async-xfer.py	2007-04-04 17:29:41.000000000 -0300
@@ -1,5 +1,5 @@
 #! /usr/bin/env python
-import gnomevfs
+from gnome import gnomevfs
 import gtk
 import termios
 import sys
Index: gnome-python-2.18/examples/vfs/async.py
===================================================================
--- gnome-python-2.18.orig/examples/vfs/async.py	2007-04-04 17:27:07.000000000 -0300
+++ gnome-python-2.18/examples/vfs/async.py	2007-04-04 17:29:41.000000000 -0300
@@ -1,6 +1,6 @@
 import pygtk; pygtk.require("2.0")
 import gtk
-import gnomevfs
+from gnome import gnomevfs
 
 def task_done():
     global counter
Index: gnome-python-2.18/examples/vfs/cancellation.py
===================================================================
--- gnome-python-2.18.orig/examples/vfs/cancellation.py	2007-04-04 17:27:07.000000000 -0300
+++ gnome-python-2.18/examples/vfs/cancellation.py	2007-04-04 17:29:41.000000000 -0300
@@ -1,6 +1,6 @@
 #! /usr/bin/env python
 import time
-import gnomevfs
+from gnome import gnomevfs
 import thread
 import sys
 
Index: gnome-python-2.18/examples/vfs/shell.py
===================================================================
--- gnome-python-2.18.orig/examples/vfs/shell.py	2007-04-04 17:27:07.000000000 -0300
+++ gnome-python-2.18/examples/vfs/shell.py	2007-04-04 17:29:41.000000000 -0300
@@ -6,7 +6,7 @@
 _normal = os.popen('tput rmso', 'r').read()
 
 import pygtk; pygtk.require("2.0")
-import gnomevfs
+from gnome import gnomevfs
 
 class Shell:
     def __init__(self, cwd=None):
Index: gnome-python-2.18/examples/vfs/sync-xfer.py
===================================================================
--- gnome-python-2.18.orig/examples/vfs/sync-xfer.py	2007-04-04 17:27:07.000000000 -0300
+++ gnome-python-2.18/examples/vfs/sync-xfer.py	2007-04-04 17:29:41.000000000 -0300
@@ -1,8 +1,10 @@
 #! /usr/bin/env python
-try:
-    import gnomevfs
-except ImportError:
-    import gnome.vfs as gnomevfs
+#try:
+#    import gnomevfs
+#except ImportError:
+#    import gnome.vfs as gnomevf
+
+from gnome import gnomevfs
 
 import atexit
 import termios
