Index: pygame/config_unix.py
===================================================================
--- pygame.orig/config_unix.py	2007-03-06 16:32:28.000000000 -0300
+++ pygame/config_unix.py	2007-03-06 16:32:52.000000000 -0300
@@ -167,15 +167,20 @@
     for d in DEPS:
         d.configure(incdirs, libdirs)
 
+# INdT
+    # Ugly hack to fix the sdl-config return in Maemo platform
+    DEPS[0].cflags = "-I/usr/X11R6/include -L/usr/X11R6/lib " + DEPS[0].cflags
 
-    for d in DEPS[1:]:
-        if not d.found:
-            if not confirm("""
-Warning, some of the pygame dependencies were not found. Pygame can still
-compile and install, but games that depend on those missing dependencies
-will not run. Would you like to continue the configuration?"""):
-                raise SystemExit
-            break
+    # Remove the manual intervention requirement
+#    for d in DEPS[1:]:
+#        if not d.found:
+#            if not confirm("""
+#Warning, some of the pygame dependencies were not found. Pygame can still
+#compile and install, but games that depend on those missing dependencies
+#will not run. Would you like to continue the configuration?"""):
+#                raise SystemExit
+#            break
+# /INdT
 
     return DEPS
 
