Index: pygame-1.7.1/Setup.in
===================================================================
--- pygame-1.7.1.orig/Setup.in	2007-06-18 11:37:06.000000000 -0300
+++ pygame-1.7.1/Setup.in	2007-06-18 11:37:19.000000000 -0300
@@ -39,7 +39,7 @@
 #SDL as a dependency. these should not be altered
 
 base src/base.c $(SDL)
-cdrom src/cdrom.c $(SDL)
+#cdrom src/cdrom.c $(SDL)
 constants src/constants.c $(SDL)
 display src/display.c $(SDL)
 event src/event.c $(SDL)
@@ -51,7 +51,7 @@
 surface src/surface.c src/alphablit.c $(SDL)
 surflock src/surflock.c $(SDL)
 time src/time.c $(SDL)
-joystick src/joystick.c $(SDL)
+#joystick src/joystick.c $(SDL)
 draw src/draw.c $(SDL)
 image src/image.c $(SDL)
 overlay src/overlay.c $(SDL)
