Index: pygobject-2.13.1/gobject/gobjectmodule.c
===================================================================
--- pygobject-2.13.1.orig/gobject/gobjectmodule.c	2007-04-30 11:51:17.000000000 -0300
+++ pygobject-2.13.1/gobject/gobjectmodule.c	2007-10-22 14:20:55.000000000 -0300
@@ -2879,6 +2879,9 @@
     
     prefix_len = strlen(strip_prefix);
     
+    if (!g_str_has_prefix(name, strip_prefix))
+        return name;
+
     /* strip off prefix from value name, while keeping it a valid
      * identifier */
     for (j = prefix_len; j >= 0; j--) {
