Index: dbus-python-0.71/dbus/dbus_bindings.pyx
===================================================================
--- dbus-python-0.71.orig/dbus/dbus_bindings.pyx	2008-01-31 17:44:32.000000000 -0300
+++ dbus-python-0.71/dbus/dbus_bindings.pyx	2008-01-31 17:49:45.000000000 -0300
@@ -247,8 +247,12 @@
     conn = Connection()
     conn.__cinit__(None, connection)
 
-    args = (conn)
-    function(*args)
+    args = (conn,)
+    try:
+        function(*args)
+    except Exception, e:
+        print "_GIL_safe_cunregister_function_handler error: %s\nAt: %s" % (e, function)
+
     Py_XDECREF(tup) 
 
 cdef void cunregister_function_handler (DBusConnection *connection,
@@ -282,7 +286,11 @@
     args = (conn,
             message)
 
-    retval = function(*args)
+    try:
+        retval = function(*args)
+    except Exception, e:
+        print "_GIL_safe_cmessage_function_handler error: %s\nAt: %s" % (e, function)
+        retval = None
 
     if (retval == None):
         retval = DBUS_HANDLER_RESULT_HANDLED
