Index: pygobject-2.16.0/tests/runtests.py
===================================================================
--- pygobject-2.16.0.orig/tests/runtests.py	2009-03-08 14:11:20.765178704 +0100
+++ pygobject-2.16.0/tests/runtests.py	2009-03-08 14:11:29.154585458 +0100
@@ -22,7 +22,7 @@ common.importModules(buildDir=buildDir,
                      srcDir=srcDir)
 
 SKIP_FILES = ['common', 'runtests',
-              'test_enum', 'test_conversion']
+              'test_enum', 'test_conversion', 'test_subtype']
 
 dir = os.path.split(os.path.abspath(__file__))[0]
 os.chdir(dir)
