Index: gtk/Makefile.am
===================================================================
--- gtk.orig/Makefile.am	2007-05-30 16:28:47.000000000 -0300
+++ gtk/Makefile.am	2007-05-30 16:28:53.000000000 -0300
@@ -130,7 +130,7 @@
 if BUILD_GTK
 pygtkexec_LTLIBRARIES += _gtk.la
 defs_DATA += $(GDK_DEFS) $(GDK_TYPES_DEFS) $(GTK_DEFS) $(GTK_TYPES_DEFS) \
-	gdk.defs gtk.defs gdk-types.defs gtk-types.defs
+	gdk.defs gtk.defs gdk-types.defs gtk-types.defs gtk-extrafuncs.defs
 endif
 
 # libglade module
