--- pyopenssl-0.6.orig/doc/pyOpenSSL.ps
+++ pyopenssl-0.6/doc/pyOpenSSL.ps
@@ -8,7 +8,7 @@
 %DVIPSWebPage: (www.radicaleye.com)
 %DVIPSCommandLine: dvips -N0 -o pyOpenSSL.ps pyOpenSSL
 %DVIPSParameters: dpi=600, compressed
-%DVIPSSource:  TeX output 2004.08.13:2054
+%DVIPSSource:  TeX output 2005.03.14:1158
 %%BeginProcSet: texc.pro
 %!
 /TeXDict 300 dict def TeXDict begin/N{def}def/B{bind def}N/S{exch}N/X{S
@@ -1651,10 +1651,10 @@
 %%Page: 1 1
 TeXDict begin 1 0 bop 0 83 3901 9 v 1890 351 a Fu(Python)64
 b(Op)5 b(enSSL)64 b(Manual)3443 520 y Ft(Release)33 b(0.6)3189
-874 y Fs(Ma)m(rtin)k(Sj\366gren)3256 1229 y Fr(13th)27
-b(August)h(2004)3301 1375 y Fq(ma)n(rtin@strakt.com)1781
-1597 y Fp(Abstract)208 1741 y Fo(This)k(mo)r(dule)f(is)h(a)g(rather)g
-(thin)f(wrapp)r(er)h(around)g(\(a)g(subset)f(of)6 b(\))32
+874 y Fs(Ma)m(rtin)k(Sj\366gren)3286 1229 y Fr(14th)28
+b(Marc)n(h)e(2005)3301 1375 y Fq(ma)n(rtin@strakt.com)1781
+1597 y Fp(Abstract)208 1741 y Fo(This)32 b(mo)r(dule)f(is)h(a)g(rather)
+g(thin)f(wrapp)r(er)h(around)g(\(a)g(subset)f(of)6 b(\))32
 b(the)g(Op)r(enSSL)e(library)-6 b(.)52 b(With)32 b(thin)f(wrapp)r(er)
 208 1833 y(I)d(mean)h(that)f(a)i(lot)f(of)h(the)f(ob)t(ject)h(metho)r
 (ds)e(do)h(nothing)g(more)g(than)g(calling)h(a)g(corresp)r(onding)g
