Index: gnome-python-2.18/gnomevfs/vfs-file-info.c
===================================================================
--- gnome-python-2.18.orig/gnomevfs/vfs-file-info.c	2007-04-04 08:49:00.000000000 -0300
+++ gnome-python-2.18/gnomevfs/vfs-file-info.c	2007-04-04 09:24:37.000000000 -0300
@@ -109,6 +109,9 @@
 	else
 	    return PyLong_FromUnsignedLong(finfo->link_count);
     } else if (!strcmp(attr, "uid")) {
+	PyErr_SetString(PyExc_ValueError, "uid field unavailable in Maemo platform");
+	return NULL;
+/*INdT - Maemo doesn't have GNOME_VFS_FILE_INFO_FIELDS_IDS
 	if (!(finfo->valid_fields & GNOME_VFS_FILE_INFO_FIELDS_IDS)) {
 	    PyErr_SetString(PyExc_ValueError, "uid field has no valid value");
 	    return NULL;
@@ -116,8 +119,11 @@
 	if (finfo->uid < G_MAXLONG)
 	    return PyInt_FromLong(finfo->uid);
 	else
-	    return PyLong_FromUnsignedLong(finfo->uid);
+	    return PyLong_FromUnsignedLong(finfo->uid);*/
     } else if (!strcmp(attr, "gid")) {
+	PyErr_SetString(PyExc_ValueError, "gid field unavailable in Maemo platform");
+	return NULL;
+/*INdT - Maemo doesn't have GNOME_VFS_FILE_INFO_FIELDS_IDS
 	if (!(finfo->valid_fields & GNOME_VFS_FILE_INFO_FIELDS_IDS)) {
 	    PyErr_SetString(PyExc_ValueError, "gid field has no valid value");
 	    return NULL;
@@ -125,7 +131,7 @@
 	if (finfo->gid < G_MAXLONG)
 	    return PyInt_FromLong(finfo->gid);
 	else
-	    return PyLong_FromUnsignedLong(finfo->gid);
+	    return PyLong_FromUnsignedLong(finfo->gid);*/
     } else if (!strcmp(attr, "size")) {
 	if (!(finfo->valid_fields & GNOME_VFS_FILE_INFO_FIELDS_SIZE)) {
 	    PyErr_SetString(PyExc_ValueError, "size field has no valid value");
@@ -314,6 +320,9 @@
         return 0;
 
     } else if (!strcmp(attr, "uid")) {
+	PyErr_SetString(PyExc_ValueError, "uid field unavailable in Maemo platform");
+	return -1;
+/*INdT - Maemo doesn't have GNOME_VFS_FILE_INFO_FIELDS_IDS
         if (!(PyInt_Check(value) || PyLong_Check(value))) {
             	PyErr_SetString(PyExc_TypeError,
                                 "'uid' attribute must be an int or long");
@@ -325,8 +334,11 @@
 	else
 	    finfo->uid = PyLong_AsUnsignedLongLong(value);
         return 0;
-
+*/
     } else if (!strcmp(attr, "gid")) {
+	PyErr_SetString(PyExc_ValueError, "gid field unavailable in Maemo platform");
+	return -1;
+/*INdT - Maemo doesn't have GNOME_VFS_FILE_INFO_FIELDS_IDS
         if (!(PyInt_Check(value) || PyLong_Check(value))) {
             	PyErr_SetString(PyExc_TypeError,
                                 "'gid' attribute must be an int or long");
@@ -338,7 +350,7 @@
 	else
 	    finfo->gid = PyLong_AsUnsignedLongLong(value);
         return 0;
-
+*/
     } else if (!strcmp(attr, "size")) {
         if (!(PyInt_Check(value) || PyLong_Check(value))) {
             	PyErr_SetString(PyExc_TypeError,
