Index: python-gstreamer/gst/gst.defs
===================================================================
--- python-gstreamer.orig/gst/gst.defs	2008-03-04 11:33:18.000000000 -0300
+++ python-gstreamer/gst/gst.defs	2008-03-04 11:33:25.000000000 -0300
@@ -2789,13 +2789,15 @@
   )
 )
 
-(define-virtual supported
-  (of-object "GstImplementsInterface")
-  (return-type "gboolean")
-  (parameters
-   '("GType" "iface_type")
-  )
-)
+; INdT
+;(define-virtual supported
+;  (of-object "GstImplementsInterface")
+;  (return-type "gboolean")
+;  (parameters
+;   '("GType" "iface_type")
+;  )
+;)
+; /INdT
 
 ;; From ../gstreamer/gst/gstiterator.h
 
@@ -6451,14 +6453,16 @@
   (return-type "GstCaps*")
 )
 
-(define-method call_function
-  (of-object "GstTypeFindFactory")
-  (c-name "gst_type_find_factory_call_function")
-  (return-type "none")
-  (parameters
-    '("GstTypeFind*" "find")
-  )
-)
+; INdT
+;(define-method call_function
+;  (of-object "GstTypeFindFactory")
+;  (c-name "gst_type_find_factory_call_function")
+;  (return-type "none")
+;  (parameters
+;    '("GstTypeFind*" "find")
+;  )
+;)
+; /INdT
 
 
 ;; From ../gstreamer/gst/gsturi.h
