Index: python-gstreamer/gst/gst-types.defs
===================================================================
--- python-gstreamer.orig/gst/gst-types.defs	2008-03-04 11:33:18.000000000 -0300
+++ python-gstreamer/gst/gst-types.defs	2008-03-04 11:33:31.000000000 -0300
@@ -311,13 +311,15 @@
   )
 )
 
-(define-pointer TypeFind
-  (in-module "Gst")
-  (c-name "GstTypeFind")
-  (gtype-id "GST_TYPE_TYPE_FIND")
-  (fields
-  )
-)
+; INdT
+;(define-pointer TypeFind
+;  (in-module "Gst")
+;  (c-name "GstTypeFind")
+;  (gtype-id "GST_TYPE_TYPE_FIND")
+;  (fields
+;  )
+;)
+; /INdT
 
 ;; Enumerations and flags ...
 
@@ -1069,10 +1071,12 @@
    (gtype-id "GST_TYPE_TAG_SETTER")
 )
 
-(define-interface ImplementsInterface
-   (in-module "Gst")
-   (c-name "GstImplementsInterface")
-   (gtype-id "GST_TYPE_IMPLEMENTS_INTERFACE")
-   (vtable "GstImplementsInterfaceClass")
-)
+; INdT
+;(define-interface ImplementsInterface
+;   (in-module "Gst")
+;   (c-name "GstImplementsInterface")
+;   (gtype-id "GST_TYPE_IMPLEMENTS_INTERFACE")
+;   (vtable "GstImplementsInterfaceClass")
+;)
+; /INdT
 
