Index: dbus-python-0.71/debian/changelog
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dbus-python-0.71/debian/changelog	2008-01-31 17:54:42.000000000 -0300
@@ -0,0 +1,103 @@
+python-dbus (0.71-1osso10) unstable; urgency=low
+
+  * Fix remove_from_connection compatibility with 0.71.
+  * Added try/except shielding to GIL_c*_function_handlers
+
+ -- Lauro Moura Maranhao Neto <lauro.neto@openbossa.org>  Thu, 31 Jan 2008 17:53:46 -0300
+
+python-dbus (0.71-1osso9) unstable; urgency=low
+
+  * Add remove_from_connection function 
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Thu, 17 Jan 2008 16:25:14 -0300
+
+python-dbus (0.71-1osso8) unstable; urgency=low
+
+  * Build python2.5-dbus-ecore package.
+
+ -- Eduardo Lima (Etrunko) <eduardo.lima@indt.org.br>  Tue, 11 Dec 2007 22:06:29 -0200
+
+python-dbus (0.71-1osso7) unstable; urgency=low
+
+  * fixed size mismatch problem at apt-get
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Thu, 25 Oct 2007 13:26:55 -0300
+
+python-dbus (0.71-1osso6) unstable; urgency=low
+
+  * fixed #972
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Wed, 24 Oct 2007 15:34:13 -0300
+
+python-dbus (0.71-1osso5) unstable; urgency=low
+
+  * added dependency to python2.5-xml (#751)
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Wed, 25 Apr 2007 15:18:00 -0300
+
+python-dbus (0.71-1osso4) unstable; urgency=low
+
+  * added a "instrospect" parameter to the get_object() method to disable the
+    Introspectable call.
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 9 Jan 2007 14:38:00 -0300
+
+python-dbus (0.71-1osso3) unstable; urgency=low
+
+  * rebuild the package after fix a problem with subversion
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri, 5 Jan 2007 11:40:00 -0300
+
+python-dbus (0.71-1osso2) unstable; urgency=low
+
+  * applied the patch in https://bugs.freedesktop.org/show_bug.cgi?id=6943 to
+    remove the libxml2 dependency
+  * create a workaround to the missing libfakeroot bug in build process
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 26 Dec 2006 15:58:48 -0300
+
+python-dbus (0.71-1osso1) unstable; urgency=low
+
+  * upgraded to python-dbus-0.71 
+
+ -- Luciano Miguel Wolf <luciano.wolf@indt.org.br>  Fri, 17 Nov 2006 18:27:00 -0300
+
+python-dbus (0.61-1osso6) unstable; urgency=low
+
+  * fixed a bug related conflicting module names with pygtk
+  * reorganizing dbus directories
+  * added pkg-config build dependency
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Mon, 17 Jul 2006 14:27:26 -0300
+
+python-dbus (0.61-1osso5) unstable; urgency=low
+
+  * minor cleanups in debian/rules
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Fri,  9 Jun 2006 16:30:57 -0300
+
+python-dbus (0.61-1osso4) unstable; urgency=low
+
+  * added shlibs dependencies
+  * enable "-mthumb" and "-msoft-abi=softfp" flags on gcc
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 6 Jun 2006 19:25:00 -0300
+
+python-dbus (0.61-1osso3) unstable; urgency=low
+
+  * added python2.4-pyrex in Build-Depends
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 1 Jun 2006 23:51:00 -0300
+
+python-dbus (0.61-1osso2) unstable; urgency=low
+
+  * remove extract.pyc file in debian/rules clean target
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 19 May 2006 13:48:00 -0300
+
+python-dbus (0.61-1osso1) unstable; urgency=low
+
+  * first version
+
+ -- Osvaldo Santana Neto <osvaldo.santana@indt.org.br>  Tue, 16 May 2006 13:48:00 -0300
+
Index: dbus-python-0.71/debian/compat
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dbus-python-0.71/debian/compat	2008-01-31 17:51:27.000000000 -0300
@@ -0,0 +1 @@
+4
Index: dbus-python-0.71/debian/control
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dbus-python-0.71/debian/control	2008-01-31 17:51:27.000000000 -0300
@@ -0,0 +1,21 @@
+Source: python-dbus
+Section: python
+Priority: optional
+Maintainer: Osvaldo Santana Neto <osvaldo.santana@indt.org.br>
+Build-Depends: debhelper (>= 4.0.0), python2.5 (>= 2.5.1-1osso2), python2.5-dev (>= 2.5.1-1osso2), libdbus-1-dev (>= 0.61), libdbus-glib-1-dev (>= 0.61), python2.5-pyrex (>= 0.9.4.1-1osso4), pkg-config, libedbus-dev
+Standards-Version: 3.6.1
+
+Package: python2.5-dbus
+Architecture: any
+Depends: python2.5 (>= 2.5.1-1osso2), python2.5-xml (>= 0.8.4-1osso8), libdbus-1-2 (>= 0.61), libdbus-glib-1-2 (>= 0.61), ${shlibs:Depends}
+Description: Python wrappers around D-Bus.
+
+Package: python2.5-dbus-ecore
+Architecture: any
+Depends: python2.5 (>= 2.5.1-1osso2), libedbus, python2.5-dbus
+Description: Python wrappers around D-Bus.
+
+Package: python2.5-dbus-dev
+Architecture: all
+Depends: python2.5-dbus (= ${Source-Version}), python2.5-xml-dev
+Description: Development files for Python wrappers around D-Bus.
Index: dbus-python-0.71/debian/copyright
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dbus-python-0.71/debian/copyright	2008-01-31 17:51:27.000000000 -0300
@@ -0,0 +1,6 @@
+This package was debianized by Osvaldo Santana Neto
+<osvaldo.santana@indt.org.br> on Tue, 16 May 2006
+
+It was downloaded from http://dbus.freedesktop.org/
+
+Copyright Holder: see COPYING
Index: dbus-python-0.71/debian/rules
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dbus-python-0.71/debian/rules	2008-01-31 17:53:02.000000000 -0300
@@ -0,0 +1,73 @@
+#!/usr/bin/make -f
+
+PYVER=2.5
+PYTHON=python$(PYVER)
+
+p_base=$(PYTHON)-dbus
+p_ecore =$(PYTHON)-dbus-ecore
+p_dev =$(p_base)-dev
+
+d_base=debian/$(p_base)
+d_ecore=debian/$(p_ecore)
+d_dev=debian/$(p_dev)
+
+ifeq ($(DEB_BUILD_ARCH),armel)
+    CFLAGS=-Os -mthumb -mfloat-abi=softfp -Wall -Wstrict-prototypes
+else
+    CFLAGS=-Os -Wall -Wstrict-prototypes
+endif
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+	CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	$(PYTHON) setup.py clean --all
+	-rm -f build-stamp
+	-rm -rf $(d_base) $(d_dev) $(d_ecore)
+	-rm -f dbus/*.py[co]
+	dh_clean 
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_installdirs
+	$(PYTHON) setup.py install --root=$(d_base) --no-compile
+	$(PYTHON) -OO -m compileall -d / -f $(d_base)
+	cp -a $(d_base)/* $(d_dev)
+	# base
+	find $(d_base) -name "*.py" -type f | xargs rm -f
+	# ecore
+	mkdir -p $(d_ecore)/usr/lib/python2.5/site-packages/dbus/
+	mv $(d_base)/usr/lib/python2.5/site-packages/dbus/*ecore* $(d_ecore)/usr/lib/python2.5/site-packages/dbus/
+	# dev
+	find $(d_dev)/usr/lib/$(PYTHON) -not -name "*.py" -type f | xargs rm -f
+	dh_installexamples -p$(p_dev) dbus/examples/*
+	# cleanup
+	find $(d_base) $(d_dev) $(d_ecore) -name ".svn" | xargs rm -rf
+
+
+binary: binary-indep binary-arch
+binary-indep: install
+binary-arch: install
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs -p$(p_dev)
+	dh_installdocs -p$(p_dev)
+	dh_installman -p$(p_dev)
+	dh_strip
+	dh_compress -p$(p_dev) -Xexamples -X.py
+	dh_fixperms
+	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+.PHONY: build install clean binary binary-indep binary-arch
+
