diff -urN --exclude='debian*' python-dbus-0.71/dbus/dbus_ecore_bindings.pyx python-dbus-0.71-dev/dbus/dbus_ecore_bindings.pyx
--- python-dbus-0.71/dbus/dbus_ecore_bindings.pyx	1969-12-31 21:00:00.000000000 -0300
+++ python-dbus-0.71-dev/dbus/dbus_ecore_bindings.pyx	2007-09-19 18:35:33.000000000 -0300
@@ -0,0 +1,15 @@
+cimport dbus_bindings
+import dbus_bindings
+
+
+cdef extern from "E_DBus.h":
+    cdef struct E_DBus_Connection:
+        pass
+
+    E_DBus_Connection *e_dbus_connection_setup(dbus_bindings.DBusConnection *conn)
+
+
+def setup_with_e_main(conn):
+    cdef dbus_bindings.Connection connection
+    connection = conn
+    e_dbus_connection_setup(connection._get_conn())
diff -urN --exclude='debian*' python-dbus-0.71/dbus/ecore.py python-dbus-0.71-dev/dbus/ecore.py
--- python-dbus-0.71/dbus/ecore.py	1969-12-31 21:00:00.000000000 -0300
+++ python-dbus-0.71-dev/dbus/ecore.py	2007-09-19 18:02:02.000000000 -0300
@@ -0,0 +1,10 @@
+import dbus
+import dbus_ecore_bindings
+
+def _setup_with_e_main(conn):
+    dbus_ecore_bindings.setup_with_e_main(conn._connection)
+
+def init_threads():
+    pass
+
+setattr(dbus, "_dbus_mainloop_setup_function", _setup_with_e_main)
Binary files python-dbus-0.71/dbus/types.pyc and python-dbus-0.71-dev/dbus/types.pyc differ
diff -urN --exclude='debian*' python-dbus-0.71/setup.py python-dbus-0.71-dev/setup.py
--- python-dbus-0.71/setup.py	2007-09-20 18:19:34.000000000 -0300
+++ python-dbus-0.71-dev/setup.py	2007-09-19 18:11:48.000000000 -0300
@@ -64,8 +64,10 @@
 includedirs_flag = ['-I.', '-Idbus/']
 dbus_includes = ['.']
 dbus_glib_includes = ['.']
+edbus_includes = ['.']
 dbus_libs = []
 dbus_glib_libs = []
+edbus_libs = []
 
 output = check_package("dbus-1", "--cflags").split()
 dbus_includes.extend([ x.replace("-I", "") for x in output ])
@@ -81,6 +83,13 @@
 output = check_package("dbus-glib-1", "--libs-only-L").split()
 dbus_glib_libs.extend([ x.replace("-L", "") for x in output ])
 
+output = check_package("edbus", "--cflags").split()
+edbus_includes.extend([ x.replace("-I", "") for x in output ])
+includedirs_flag.extend(output)
+
+output = check_package("edbus", "--libs-only-L").split()
+edbus_libs.extend([ x.replace("-L", "") for x in output ])
+
 output = open("dbus/dbus_bindings.pxd", 'w')
 extract.main("dbus/dbus_bindings.pxd.in", includedirs_flag, output)
 output.close()
@@ -123,6 +132,7 @@
         "dbus/_dbus",
         "dbus/exceptions",
         "dbus/glib",
+        "dbus/ecore",
         "dbus/__init__",
         "dbus/matchrules",
         "dbus/service",
@@ -147,6 +157,11 @@
                 ('DBUS_API_SUBJECT_TO_CHANGE', '1')
             ],
         ),
+        Extension("dbus/dbus_ecore_bindings", ["dbus/dbus_ecore_bindings.pyx"],
+            include_dirs=edbus_includes,
+            library_dirs=edbus_libs,
+            libraries=["ecore", "dbus-1", "edbus"],
+        ),
     ],
     cmdclass={'build_ext': build_ext, 'clean': full_clean, 'check': dbus_python_check}
 )
