Index: dbus-python-0.71/dbus/dbus_bindings.pyx
===================================================================
--- dbus-python-0.71.orig/dbus/dbus_bindings.pyx	2008-01-31 17:37:33.000000000 -0300
+++ dbus-python-0.71/dbus/dbus_bindings.pyx	2008-01-31 17:43:40.000000000 -0300
@@ -281,7 +281,7 @@
     conn.__cinit__(None, connection)
     args = (conn,
             message)
- 
+
     retval = function(*args)
 
     if (retval == None):
@@ -334,6 +334,9 @@
     cdef _set_conn(self, DBusConnection *conn):
         self.conn = conn
     
+    def _unregister_object_path(self, path):
+        return dbus_connection_unregister_object_path(self.conn, path)
+
     cdef DBusConnection *_get_conn(self):
         return self.conn
     
Index: dbus-python-0.71/dbus/service.py
===================================================================
--- dbus-python-0.71.orig/dbus/service.py	2008-01-31 17:37:33.000000000 -0300
+++ dbus-python-0.71/dbus/service.py	2008-01-31 17:40:03.000000000 -0300
@@ -270,6 +270,21 @@
 
         self._connection.register_object_path(object_path, self._unregister_cb, self._message_cb)
 
+    @property 
+    def __dbus_object_path__(self):
+        return self._object_path
+
+    @property 
+    def connection(self):
+        return self._connection 
+
+    def remove_from_connection(self, connection=None, path=None):
+        try:
+            self._connection._unregister_object_path(self._object_path)
+        finally:
+            self._connection = None
+            self._object_path = None
+    
     def _unregister_cb(self, connection):
         print ("Unregister")
 
