Index: pybluez/doc/gendoc
===================================================================
--- pybluez.orig/doc/gendoc	2007-03-06 15:39:51.000000000 -0300
+++ pybluez/doc/gendoc	2007-03-06 15:40:11.000000000 -0300
@@ -8,8 +8,8 @@
 for d in $gendocdir/../build/lib*/ 
 do
     cd $d
-    pydoc -w bluetooth >> /dev/null
-    pydoc -w _bluetooth >> /dev/null
+    pydoc2.5 -w bluetooth >> /dev/null
+    pydoc2.5 -w _bluetooth >> /dev/null
     mv -f bluetooth.html ../../doc/bluetooth.html
     mv -f _bluetooth.html ../../doc/_bluetooth.html
     break
