Index: python-gtk2-2.12.0/codegen/h2def.py
===================================================================
--- python-gtk2-2.12.0.orig/codegen/h2def.py	2007-09-15 12:00:49.000000000 -0300
+++ python-gtk2-2.12.0/codegen/h2def.py	2008-01-08 15:31:30.000000000 -0300
@@ -89,7 +89,7 @@
     while pos < len(buf):
         m = pat.search(buf, pos)
         if not m: break
-        maybeobjdefs.append((m.group(2), m.group(2)))
+        maybeobjdefs.append((m.group(2), m.group(1)))
         pos = m.end()
 
     # now find all structures that look like they might represent a class:
