Index: python-numeric/Misc/lapack_lite/clapack_scrub.py
===================================================================
--- python-numeric.orig/Misc/lapack_lite/clapack_scrub.py	2007-03-06 18:02:10.000000000 -0300
+++ python-numeric/Misc/lapack_lite/clapack_scrub.py	2007-03-06 18:02:21.000000000 -0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.4
+#!/usr/bin/env python2.5
 
 import sys, os
 from cStringIO import StringIO
